R/plotting.R

```diff
@@ -18,6 +18,47 @@
 # suppressPackageStartupMessages(library(d3r))
 # suppressPackageStartupMessages(library(viridis))
 
+########################
+## Internal Functions ##
+########################
+#' 
+#' 
+.LevelReduction <- function(lin, level) {   
+    if (level == 1) {
+        gt_loc <- str_locate(lin, ">")[[1]]
+        if (is.na(gt_loc)) {
+            # No '>' in lineage
+            return(lin)
+        } else {
+            lin <- substring(lin, first = 0, last = (gt_loc - 1))
+            return(lin)
+        }
+    }
+    # Out of bounds guard
+    gt_loc <- str_locate_all(lin, ">")[[1]] 
+    l <- length(gt_loc) / 2
+    if (level > l) {
+        # Not enough '>' in lineage
+        return(lin)
+    } else {
+        gt_loc <- gt_loc[level, ][1] %>% as.numeric()
+        lin <- substring(lin, first = 0, last = (gt_loc - 1))
+        return(lin)
+    }
+}
+
+.GetKingdom <- function(lin) {
```

Comment:
Function Modularization:

Functions .LevelReduction and .GetKingdom, initially defined within plotLineageHeatmap, are moved outside and set as independent internal functions. This modularization improves reusability, readability, and reduces nested function complexity within plotLineageHeatmap.